src/Agda2Hs/Compile/Type.hs

```diff
@@ -140,7 +140,11 @@ compileType t = do
       x  <- hsName <$> compileVar x
       return $ tApp (Hs.TyVar () x) vs
     Sort s -> return (Hs.TyStar ())
-    t -> genericDocError =<< text "Bad Haskell type:" <?> prettyTCM t
+    Lam argInfo restAbs
+      | usableModality argInfo  -> genericDocError =<< text "Bad Haskell type:" <?> prettyTCM t
```

Comment:
You can leave out this line and just let it fall through to the `_ ->` case below

Reply:
Also, please use the `keepArg` utility function (https://github.com/agda/agda2hs/blob/fcd2885e3775cfb96cf100303510f7485fb5e82c/src/Agda2Hs/Compile/Utils.hs#L120) instead of `usableModality`.

Reply:
Done:)